### ISSUE TITLE
Worker: Batch classification predictions for GPU efficiency

### ISSUE THREAD

**Issue by mihow** [RolnickLab/ami-data-companion#103]
## Summary

The worker currently classifies each detection crop one at a time in a loop, which is inefficient on GPU. Crops can be batched together for a single forward pass.

Question: the classification step in the existing dataloader & classification pipeline already batches crops (defaults to 100 or 200). How does it do it without resizing or padding? Review the existing pipeline (the one that reads from the local DB queue and does batch processing). 

## Current behavior

In `trapdata/cli/worker.py:212-229`, each detection is processed individually:

```python
for idx, dresp in enumerate(detector.results):
    # ... crop the image ...
    crop = crop.unsqueeze(0)  # batch size = 1
    classifier_out = classifier.predict_batch(crop)  # single item
    # ...
```

This causes:
- High per-call overhead (GPU memory transfers, kernel launches)
- Poor GPU utilization with batch size = 1
- Slower processing, especially for images with many detections

## Proposed solution

Collect all crops first, then classify in a single batched call:

```python
# Collect all crops
crops = []
for dresp in detector.results:
    crop = image_tensor[:, int(bbox.y1):int(bbox.y2), int(bbox.x1):int(bbox.x2)]
    crops.append(crop)

# Batch and classify all at once
if crops:
    # May need padding/resizing for uniform tensor sizes
    batched_crops = torch.stack(crops)  
    classifier_out = classifier.predict_batch(batched_crops)
    # ... map predictions back to detections ...
```

## Considerations

- Crops may have different sizes - need to resize or pad to uniform dimensions
- Memory constraints - may need to sub-batch if too many detections
- Need benchmarking to measure actual improvement
- Should verify classification accuracy is unchanged after batching

## Related

- Flagged by Copilot in PR #94 review
- CLAUDE.md notes "Batch processing is critical"

## Test plan

- [ ] Benchmark current vs batched implementation on sample images
- [ ] Verify classification results are identical
- [ ] Test with images containing many detections (10+, 50+)
- [ ] Test memory usage with large batches
